src/core/lb/LBWalberla.cpp

```diff
@@ -112,6 +114,30 @@ void LBWalberla::sanity_checks(System::System const &system) const {
                              system.get_time_step());
 }
 
+void LBWalberla::on_lees_edwards_change() {
+  auto &system = System::get_system();
+  auto const energy_conversion =
+      Utils::int_pow<2>(lb_params->get_agrid() / lb_params->get_tau());
+  auto const lb_kT = lb_fluid->get_kT() * energy_conversion;
+  auto le_protocol = system.lees_edwards->get_protocol();
+  if (lb_kT != 0.) {
+    throw std::runtime_error("Lees-Edwards LB doesn't support thermalization");
+  }
+  auto const &le_bc = system.box_geo->lees_edwards_bc();
+  auto lees_edwards_object = std::make_unique<LeesEdwardsPack>(
+      le_bc.shear_direction, le_bc.shear_plane_normal,
```

Comment:
I had a quick look at the Lees-Edwards code recently for an unrelated task, and vaguely remember that there are two ways for the user to remove Lees-Edwards: either set the `Off` protocol, or assign the Python value `None`, in which case variable `le_protocol` at line 122 in this file is a nullptr. It was quite a headache for @1234somesh to find all the places in the code where he had to introduce conditional checks for nullptr.

I'm mentioning this, because the script interface doesn't seem to call `system.on_lees_edwards_change()` when users set the protocol to `None`, in which case the LB keeps the LE collision operator. Could you please double-check that edge case? If you determine there is a need to call `system.on_lees_edwards_change()` when taking the `if (is_none(protocol))` branch in `src/script_interface/lees_edwards/LeesEdwards.hpp` (to reproduce the same behavior as setting the protocol to `Off`), then a nullptr check will be necessary in this function. An elegant way to introduce such a nullptr check will be described in the next code review comment, where an `update_collision_model()` method is proposed.